sdk/src/types/block/output/feature/mod.rs

```diff
@@ -143,6 +153,21 @@ impl Feature {
             panic!("as_tag called on a non-tag feature");
         }
     }
+
+    /// Checks whether the feature is a [`StakingFeature`].
+    pub fn is_staking(&self) -> bool {
+        matches!(self, Self::Staking(_))
+    }
+
+    /// Gets the feature as an actual [`StakingFeature`].
+    /// PANIC: do not call on a non-staking feature.
+    pub fn as_staking(&self) -> &StakingFeature {
+        if let Self::Staking(feature) = self {
+            feature
+        } else {
+            panic!("as_staking called on a non-staking feature");
```

Comment:
I feel the messaging could be better. The doc should say something like

```rust
/// NOTE: Will panic if the feature is not a `StakingFeature`.
```

and the panic message like


```rust
panic!("invalid downcast of non-staking feature");
```

Reply:
I'll change that in develop as well

Reply:
Done